test/e2e/prometheusagent_test.go

```diff
@@ -365,3 +374,240 @@ func testPromAgentReconcileDaemonSetResourceDelete(t *testing.T) {
 	require.NoError(t, pollErr)
 	require.NoError(t, err)
 }
+
+func testPrometheusAgentDaemonSetSelectPodMonitor(t *testing.T) {
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ctx := context.Background()
+	name := "test"
+
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+	_, err := framework.CreateOrUpdatePrometheusOperatorWithOpts(
+		ctx, testFramework.PrometheusOperatorOpts{
+			Namespace:           ns,
+			AllowedNamespaces:   []string{ns},
+			EnabledFeatureGates: []string{"PrometheusAgentDaemonSet"},
+		},
+	)
+	require.NoError(t, err)
+
+	app, err := testFramework.MakeDeployment("../../test/framework/resources/basic-app-for-daemonset-test.yaml")
+	require.NoError(t, err)
+
+	err = framework.CreateDeployment(ctx, ns, app)
+	require.NoError(t, err)
+
+	pm := framework.MakeBasicPodMonitor(name)
+	pm.Spec.PodMetricsEndpoints = []monitoringv1.PodMetricsEndpoint{monitoringv1.PodMetricsEndpoint{
+		Port: "web",
+		BasicAuth: &monitoringv1.BasicAuth{
+			Username: v1.SecretKeySelector{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: name,
+				},
+				Key: "user",
+			},
+			Password: v1.SecretKeySelector{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: name,
+				},
+				Key: "password",
+			},
+		},
+	}}
+	_, err = framework.MonClientV1.PodMonitors(ns).Create(ctx, pm, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	prometheusAgentDS := framework.MakeBasicPrometheusAgentDaemonSet(ns, name)
+	prometheusAgentDS.Spec.ScrapeInterval = "1s"
+
+	_, err = framework.CreatePrometheusAgentAndWaitUntilReady(ctx, ns, prometheusAgentDS)
+	require.NoError(t, err)
+
+	var pollErr error
+	err = wait.PollUntilContextTimeout(context.Background(), 20*time.Second, 20*time.Minute, false, func(ctx context.Context) (bool, error) {
+		appPods, err := framework.KubeClient.CoreV1().Pods(ns).List(ctx, metav1.ListOptions{
+			LabelSelector: "group=test",
+		})
+		if err != nil {
+			pollErr = fmt.Errorf("can't list app pods: %w", err)
```

Comment:
@simonpasquier @ArthurSens The test is failing here. Can you help me figure out?